src/tests/ftest/control/dmg_pool_query_ranks.py

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/python3
+"""
+  (C) Copyright 2022 Intel Corporation.
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+"""
+import time
+
+from control_test_base import ControlTestBase
+
+
+class DmgPoolQueryRanks(ControlTestBase):
+    # pylint: disable=too-many-ancestors
+    """Test dmg query ranks enabled/disabled command.
+
+    Test Class Description:
+        Simple test to verify the pool query command ranks of dmg tool.
+
+    :avocado: recursive
+    """
+
+    def setUp(self):
+        """Set up for dmg pool query."""
+        super().setUp()
+
+        # Init the pool
+        self.add_pool(connect=False)
+
+    def test_pool_query_ranks_basic(self):
+        """Test the state of ranks with dmg pool query
+
+        Test Description:
+            Create a pool with some engines and check they are all enabled.
+
+        :avocado: tags=all,daily_regression
+        :avocado: tags=vm
+        :avocado: tags=dmg,pool_query,pool_query_ranks
+        :avocado: tags=pool_query_ranks_basic
+        """
+        self.log.info("Basic tests of pool query with ranks state")
+
+        self.log.debug("Checking without ranks state information")
+        data = self.dmg.pool_query(self.pool.identifier)
+        self.assertIsNone(data['response']['enabled_ranks'],
+                "Invalid enabled_ranks field: want=None, "
+                "got={}".format(data['response']['enabled_ranks']))
+        self.assertIsNone(data['response']['disabled_ranks'],
+                "Invalid disabled_ranks field: want=None, "
+                "got={}".format(data['response']['disabled_ranks']))
+
+        self.log.debug("Checking enabled ranks state information")
+        data = self.dmg.pool_query(self.pool.identifier, show_enabled=True)
+        self.assertListEqual(data['response']['enabled_ranks'], [0, 1, 2],
+                "Invalid enabled_ranks field: want=None, "
```

Comment:
The text doesn't seem to match the actual comparison here.

Reply:
- [ ] Fix invalid assertion comment